simplified-books-database/src/main/java/org/nypl/simplified/books/book_database/DatabaseFormatHandleAudioBook.kt

```diff
@@ -133,7 +137,17 @@ internal class DatabaseFormatHandleAudioBook internal constructor(
     val newFormat = synchronized(this.dataLock) {
       FileUtilities.fileDelete(this.filePosition)
 
-      this.formatRef = this.formatRef.copy(position = null)
+      if (this.fileBook.isDirectory) {
+        DirectoryUtilities.directoryDelete(this.fileBook)
+      } else {
+        FileUtilities.fileDelete(this.fileBook)
+      }
+
+      this.formatRef = this.formatRef.copy(
+        file = null,
+        position = null
+      )
+
       this.formatRef
```

Comment:
You can just remove the assignment and the return value in the end and just leave the

```
this.formatRef.copy(
        file = null,
        position = null
      )
```